src/publications_embedding.py

```diff
@@ -90,40 +119,119 @@ def embeddings_from_pubs(pubs: list[str],
 
     return embeddings
 
+def write_hdf5(hdf5_file: str,
+               embeddings: np.ndarray,
+               author_ids: np.ndarray,
+               update: bool = False):
+    
+    if not update:  # write new hdf5 file
+        with h5py.File(hdf5_file, 'w') as f:
+            f.create_dataset(name = 'publication_embedding', 
+                             data = embeddings,
+                             maxshape = (None, None),  # dataset resizable
+                             compression = 'gzip')
+            f.create_dataset(name = 'author_ids', 
+                             data = author_ids,
+                             maxshape = (None,),  # dataset resizable
+                             compression = 'gzip')
+
+    else:  # update existing hdf5 file
+        with h5py.File(hdf5_file, 'r+') as f:
+            pub_embeddings = f['publication_embedding']
+            ids = f['author_ids']
+            for id, embedding in zip(author_ids, embeddings):
+                # check if id is in dataset
+                index = np.where(ids[:] == id)[0]
+                if len(index) != 0:  # overwrite old embedding with the new one
+                    index = index[0]
```

Comment:
```suggestion
                matching_entries = np.asarray(ids[:] == id).nonzero()[0]
                if len(matching_entries) > 1:
                    print ("WARNING: There are multiple entries for author ", id, file=sys.stderr)

                if len(matching_entries) != 0:  # overwrite old embedding with the new one
                    index = matching_entries[0]
```

Reply:
Make some sanity check on the uniqueness of ids whose embedding should be replaced.